### DISCUSSION TITLE
Wpf - Browser resize sluggish

### DISCUSSION THREAD

**Discussion by touchwindow-dev** [cefsharp/CefSharp#4392] [General]
Hi everyone,

we are facing very slow and sluggish resizing performance with the wpf cefsharp control (latest stable version 109).
As you can see in the video attached the webbrowser is inside an usercontrol that is resized using touch manupulation.

Is there a way to get faster resizing or synchronizing the control and the browser?

The issue seems related to the windowless framerate parameter, if it is set to something slow like 5 the lag get much worse but we can't above 60 unfortunately.

Any help?

Thanks

https://user-images.githubusercontent.com/29534323/217505805-a7d926dd-ad00-492f-b775-3132909a74f6.mp4

**Comment by aimcosoftware** [comment]
That looks perfectly normal to me. A Google maps page in Chrome responsive design mode, is about the same speed. There's a lot of network traffic on resize.
On way to make it smoother would be to use a resizing frame, and resize the browser on release of that.

**Comment by amaitland** [comment]
In future if you open a `Discussion` (https://github.com/cefsharp/CefSharp/discussions/4387) then **please wait for a reply** before creating a duplicate issue. You are just creating extra work. I've converted this to a discussion also. Issue https://github.com/cefsharp/CefSharp/issues/2275 is currently being used to track a potential `WPF` performance improvement. If you are targeting an older version of `.Net` e.g. `4.8` then you may see performance improvements upgrading to `.Net 6/7` as the `WPF` team have been hard at work on the framework itself.

> we are facing very slow and sluggish resizing performance

As @aimcosoftware has pointed out `Google Maps` will have different performance characteristics as it's rendering is network dependant. Network conditions will also influence performance.

> The issue seems related to the windowless framerate parameter, if it is set to something slow like 5 the lag get much worse but we can't above 60 unfortunately.

Depending on your hardware/environment then `60fps` maybe the maximum. 

Make sure you have an [app.manifest](https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-4.x#40-add-appmanifest-to-your-application) with compatibility entries to ensure `GPU Acceleration` works correctly.

You can try [disable-gpu-vsync](https://peter.sh/experiments/chromium-command-line-switches/#disable-gpu-vsync) command line arg. There are also some command line args shown in the source that might yield a performance increase in some scenarios.

https://github.com/cefsharp/CefSharp/blob/7edc164206a5aea28918ecca0d115d20df4934b3/CefSharp.Core/CefSettingsBase.cs#L446

See https://github.com/cefsharp/CefSharp/wiki/General-Usage#offscreen-rendering-osr for a brief explanation of how the `WPF` control is implemented.

If you want to have a look at the rendering implementation yourself the current default implementation is [WritableBitmapRenderHandler](https://github.com/cefsharp/CefSharp/blob/cefsharp/110/CefSharp.Wpf/Rendering/WritableBitmapRenderHandler.cs). You can implement your own render handler if required.
